scripts/autodoc

```diff
@@ -1,5 +1,159 @@
+#!/bin/bash
+
+DOCS_REPO=$1
+DOCS_VERSION=$2
+
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+function echoerr() {
+    printf '%s%s%s\n' "$RED" "$@" "$NC" 1>&2;
+}
+
+function copy() {
+    printf 'cp %s\t→\t%s' "$1" "$2"
+    cp "$1" "$2"
+}
+
+INFINITY=9223372036854775807
+COUNT_INDENTS=0
+function count_indents_on_line() {
+    local file=$1
+    local lineno=$2
+    local line
+    line=$(sed -n "${lineno}p" "$file")
+    local trim="${line#*[^[:blank:]]}"
+    COUNT_INDENTS="$(( ${#line} - ${#trim} - 1 ))"
+
+    # empty lines count as "infinitely indented"
+    if [ "$COUNT_INDENTS" -lt 0 ]
+    then
+        COUNT_INDENTS="$INFINITY"
+    fi
+}
+
+function insert_pdk_nav_file_into_docs_nav_file() {
+    local pdk_nav_file=$1
+    local docs_nav_file=$2
+
+    # find first_line and last line of the current pdk index in docs_nav_file
+    # first line contains "Plugin Development Kit"
+    local first_line
+    first_line=$(sed -n "/text: Plugin Development Kit/=" "$docs_nav_file")
+
+    # find last_line by looking at the indentation of all subsequent lines, looking at the indentation
+    # last line is the last one with more indentation that first_line
+    local max_line
+    max_line=$(wc -l < "$docs_nav_file" | bc)
+    count_indents_on_line "$docs_nav_file" "$first_line"
+    local initial_indent="$COUNT_INDENTS"
+    local current_indent="$(( initial_indent + 1 ))"
+    local current_line="$first_line"
+    while [ "$current_indent" -gt "$initial_indent" ] && [ "$current_line" -lt "$max_line" ]
+    do
+        current_line="$((current_line + 1))"
+        count_indents_on_line "$docs_nav_file" "$current_line"
+        current_indent="$COUNT_INDENTS"
+    done
+    local last_line="$(( current_line - 1 ))"
+
+    # insert_line is where we want to put the new pdk docs after deleting the old ones (one less than first)
+    local insert_line="$(( first_line - 1 ))"
+
+    # delete existing pdk index, insert new pdk nav file instead
+    sed -i '' -e "${first_line},${last_line}d" -e "${insert_line}r${pdk_nav_file}" "$docs_nav_file"
```

Comment:
It has been pointed out that in general, `sed -i ''` (modifying a file in-place, without making a backup copy) is undesirable - so much that it is spelled that way in the sed manual.

I propose that in this particular application that is an ok thing to do because the backup copy already exists: the docs repo is a git repository so we can always `git checkout` to our heart's content.

(Also, in my opinion the sed manual page needs less judgement and more examples)